types.go

```diff
@@ -108,20 +108,16 @@ type FunctionInfo struct {
 
 // functionInfo contains information about which interfaces the objective
 // function F implements and the actual methods of F that have been
-// successfully type switched.
+// successfully type switched. TODO(jds) change this comment
 type functionInfo struct {
```

Comment:
I think this struct existed so that we didn't have to typeswitch on Function all the time. Now that we are explicit, this struct is probably unnecessary. And we should check if FunctionInfo (possibly renamed to ProblemInfo) is absolutely necessary or if it could be removed as well. @btracey explained some time ago why the optimizers could want it, but I forgot the reason and the package has changed since then.